src/core/Header/HeaderButton.tsx

```diff
@@ -78,11 +98,13 @@ export const HeaderButton = (props: HeaderButtonProps) => {
     ...rest,
   };
 
-  return menuItems ? (
-    <DropdownButton {...buttonProps} menuItems={menuItems} />
-  ) : (
-    <Button {...buttonProps} />
-  );
+  if (isSplitButton(buttonProps)) {
```

Comment:
@mayank99 Did you have in mind something like this?

Reply:
Yeah that looks right